velox/functions/prestosql/tests/TimestampWithTimeZoneCastTest.cpp

```diff
@@ -0,0 +1,79 @@
+/*
+ * Copyright (c) Facebook, Inc. and its affiliates.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include "velox/functions/prestosql/tests/CastBaseTest.h"
+#include "velox/functions/prestosql/types/TimestampWithTimeZoneType.h"
+
+using namespace facebook::velox;
+
+class TimestampWithTimeZoneCastTest : public functions::test::CastBaseTest {
+ public:
+  RowVectorPtr makeTimestampWithTimeZoneVector(
+      const VectorPtr& timestamps,
+      const VectorPtr& timezones) {
+    VELOX_CHECK_EQ(timestamps->size(), timezones->size());
+    return std::make_shared<RowVector>(
+        pool(),
+        TIMESTAMP_WITH_TIME_ZONE(),
+        nullptr,
+        timestamps->size(),
+        std::vector<VectorPtr>({timestamps, timezones}));
+  }
+};
+
+TEST_F(TimestampWithTimeZoneCastTest, toTimestampWithTimeZone) {
+  const auto tsVector = makeNullableFlatVector<Timestamp>(
+      {Timestamp(1996, 0), std::nullopt, Timestamp(19920, 0)});
+  const auto expected = makeTimestampWithTimeZoneVector(
+      makeFlatVector<int64_t>(
+          {1996 * kMillisInSecond, 0, 19920 * kMillisInSecond}),
+      makeFlatVector<int16_t>({0, 0, 0}));
+  expected->setNull(1, true);
+
+  evaluateAndVerify<ComplexType>(
```

Comment:
That makes sense. Or even have one helper function have both cases covered.

Reply:
> Or even have one helper function have both cases covered.

Even better.

Reply:
There is a `testCast` API in CastBaseTest.h that calls evaluateAndVerify() on flat-, constant-, and dictionary-encoded inputs. It tests both cast() and try_cast(), but doesn't test try(cast()). Also, it requires both cast() and try_cast() to have the same results because it uses the same expected result vector. It would be nice to extend CastBaseTest to have separate APIs for cast(), try_cast(), and try(cast()).

Reply:
It seems `testCast` is what needed here. Thanks!